design/providerid_format_change.md

```diff
@@ -0,0 +1,173 @@
+
+### Non-Goals
+
+Making templates simpler.
+
+## Proposal
+
+The issues discussed above can be solved by introducing a new field in the
+metal3datatemplate implicitly. This does not require a change in the API
+(metal3datatemplate object) if CAPM3 always automatically populates this key in
+the resulting secret for the node. But it should be a default only, so that it
+could be overwritten by the user if they define this key explicitly.
+
+```bash
+providerid: "metal3://<namespace>/<BareMetalHost name>/<metal3machine name>"
+```
+
+The new format is both dynamic and known in advance. Since this is added from
+the code, there is no need of changing Spec or Status sections of the resource.
+
+Also, since it is constructed in the code, there is no need for the users to
+provide provider-id and label via templates. However, the users still refer to
+the providerID variable in their templates to get it populated by cloud-init.
+
+### User Stories
+
+- As a user, I want my nodes' providerIDs to be unique per kubernetes deployment
+ and be generated automatically by CAPM3 using metal3machine and baremetalhost resource names.
+
+## Design Details
+
+The following changes are required in the code:
+
+- Searching nodes is no longer done using labels
+- Searching nodes is done using implicitly set providerid
+- Providerid field is introduced in metal3datatemplate in the code.
+- Labels are no longer provided via templates
+
+Concerning backward compatibility, there are three scenarios:
+
+1. backward compatibility with legacy providerID in existing deployments,
+  - During scale out, new nodes keep using legacy providerID
+  - During upgrade, upgraded nodes keep using legacy providerID
+2. backwards compatibility with legacy providerID for new deployments,
+  - During provisioning, if user supplies template with legacy providerID and
+  node label, then the nodes should be provisioned with the legacy providerID
+  - This will be supported for the next two releases
+
+3. transition from legacy to new providerID format for existing deployments.
+  - Existing nodes with legacy providerID can be moved to the new format using
+  upgrade. This can be achieved by creating new
+  `kubeadmconfigtemplates.bootstrap.cluster.x-k8s.io` resource with the new provider-id
+  set inside the `.spec.template.spec.joinConfiguration.nodeRegistration.kubeletExtraArgs`
+  structure. An example is shown below:
+
+```yaml
+apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+kind: KubeadmConfigTemplate
+metadata:
+  name: test1-s54fx
+  namespace: metal3
+spec:
+  clusterConfiguration: ....
+  files: ...
+  joinConfiguration:
+    controlPlane:
+      localAPIEndpoint: {}
+    discovery: ...
+    nodeRegistration:
+      kubeletExtraArgs:
+        provider-id: metal3://{{ ds.meta_data.providerid }}
+      name: '{{ ds.meta_data.name }}'
+```
+
+### Implementation Details/Notes/Constraints
+
+None
+
+### Risks and Mitigations
+
+None
+
+### Work Items
+
+None
```

Comment:
No work items? I think some of the notes from the ## Design Details can be moved here and for reference/get a better idea what to put on this section, you can check other design docs available, i.e https://github.com/metal3-io/metal3-docs/blob/main/design/baremetal-operator/bios-config.md#work-items